lib/team_statistics.rb

```diff
@@ -0,0 +1,200 @@
+require_relative "game_data"
+require_relative "team_data"
+require_relative "game_team_data"
+require_relative "league_statistics"
+require "csv"
+
+class TeamStatistics < LeagueStatistics
+
+  def initialize
+    @team_info_by_id = Hash.new
+    @by_team_id_game_objects = []
+    @total_games_by_season = Hash.new{ |hash, key| hash[key] = 0 }
+    @total_wins_by_season = Hash.new{ |hash, key| hash[key] = 0 }
+    @win_percentage_by_season = Hash.new
+    @games_won_by_team_id = Hash.new{ |hash, key| hash[key] = 0 }
+    @goals_by_game = []
+    @team_name_by_id = Hash.new{}
+    @games_played_by_opponent = Hash.new{ |hash, key| hash[key] = 0 }
+    @games_won_by_opponent = Hash.new{ |hash, key| hash[key] = 0 }
+    @win_ratio_by_opponent = Hash.new
+  end
+
+  def all_teams
+    TeamData.create_objects
+  end
+
+  def all_games
+    GameData.create_objects
+  end
+
+  def all_game_teams
+    GameTeamData.create_objects
+  end
+
+  def team_info(passed_id)
+    all_teams.each do |team|
+      if passed_id == team.team_id.to_s # Remove .to_s when spec harness info updates
+        @team_info_by_id[:team_id] = team.team_id
+        @team_info_by_id[:franchise_id] = team.franchise_id
+        @team_info_by_id[:team_name] = team.team_name
+        @team_info_by_id[:abbreviation] = team.abbreviation
+        @team_info_by_id[:link] = team.link
+      end
+    end
+    @team_info_by_id
```

Comment:
Do we need this line?

Reply:
Ooh, good catch! Will update

Reply:
Okay, tested it. It is actually needed for this method because we do want the final return to be the hash of @team_info_by_id